application/cms/classification_service.py

```diff
@@ -0,0 +1,219 @@
+            return classification_value
+
+    def update_value_position(self, value_string, value_position):
+        classification_value = self.get_value(value=value_string)
+        if classification_value:
+            classification_value.position = value_position
+            db.session.add(classification_value)
+            db.session.commit()
+            return classification_value
+        return None
+
+    @staticmethod
+    def clean_value_database():
+        values = Ethnicity.query.all()
+        for value in values:
+            if len(value.classifications) == 0:
+                db.session.delete(value)
+                db.session.commit()
+
+    """
+    CATEGORY >-< VALUE relationship management
+    """
+
+    def add_value_to_classification(self, classification, value_title):
+        value = self.get_or_create_value(value_string=value_title)
+        classification.ethnicities.append(value)
+
+        db.session.add(classification)
+        db.session.commit()
+        return classification
+
+    def add_value_to_classification_as_parent(self, classification, value_string):
+        value = self.get_or_create_value(value_string=value_string)
+        classification.parent_values.append(value)
+
+        db.session.add(classification)
+        db.session.commit()
+        return classification
+
+    def add_values_to_classification(self, classification, value_strings):
+        for value_string in value_strings:
+            value = self.get_or_create_value(value_string=value_string)
+            classification.ethnicities.append(value)
+        db.session.add(classification)
+        db.session.commit()
+        return classification
+
+    def add_values_to_classification_as_parents(self, classification, value_strings):
+        for value_string in value_strings:
+            value = self.get_or_create_value(value_string=value_string)
+            classification.parent_values.append(value)
+        db.session.add(classification)
+        db.session.commit()
+        return classification
+
+    def remove_value_from_classification(self, classification, value_string):
+        value = self.get_value(value=value_string)
+
+        classification.ethnicities.remove(value)
+        db.session.add(classification)
+        db.session.commit()
+
+    def remove_parent_value_from_classification(self, classification, value_string):
+        value = self.get_value(value=value_string)
+
+        classification.parent_values.remove(value)
+        db.session.add(classification)
+        db.session.commit()
+
+    @staticmethod
+    def remove_classification_values(classification):
+        for value in classification.ethnicities:
+            db.session.delete(value)
+        db.session.commit()
+
+    @staticmethod
+    def remove_parent_classification_values(classification):
+        for value in classification.parent_values:
+            db.session.delete(value)
+        db.session.commit()
+
+
+class ClassificationLink:
```

Comment:
We need a new name for this. And as a knock on we'll also need a new name for `EthnicityClassificationLinkBuilder`.  And `ExternalClassificationFinderLink`... because who knows what that means?

Reply:
@TheDoubleK `ClassificationAndIncludesParentsAllUnknown`? (Hopefully we'll be able to refactor this class away at some point soon)